src/worker_utils.ts

```diff
@@ -0,0 +1,62 @@
+import { Pool, PoolClient } from "pg";
+import { WorkerUtilsOptions, TaskOptions, Job } from "./interfaces";
+import { makeWithPgClientFromPool, makeAddJob } from "./helpers";
+import { migrate } from "./migrate";
+import { defaultLogger } from "./logger";
+import { withReleasers, assertPool } from "./runner";
+
+const processPublisherOptions = async (options: WorkerUtilsOptions) => {
+  const { logger = defaultLogger } = options;
+  return withReleasers(async (releasers, _) => {
```

Comment:
The second argument is the `release` method; it has to be called at some point but it's being discarded here. I think you might want to return it here so you can call it from the `end` method instead.
```suggestion
  return withReleasers(async (releasers, release) => {
```